### PR TITLE
Full Rewrite with Linux and (theoretically) MacOS Support

### PR DESCRIPTION
**Pull request by ArchLeaders**
I was bored, and I thought this was a cool tool that could be improved. So I rewrote it using AvaloniaUI (cross-platform UI framework) and .NET 6. It reflects the WinForms version, but with some QoL improvements like drag/drop and quick convert options.

### Changelog:

- Cross-platform support for the UI and Console applications (workflows only build for Windows/Linux, macOS _should_ also be supported though)
- DotNET 6 and C# 10 features and performance
- General UI/UX improvement
  - Dark theme (light theme is also supported, but not implemented)
  - Quicker save convert menu item `File > Convert Save`
  - Folder drag/drop support
  - Built-in async logs viewer
- Refactored save converter (Switch saves for BOTW v1.6 were not tested)
- Improved logging
- Clarified documentation (ReadMe)
- GitHub publishing workflows (trigger with `git tag vX.Y.X && git push --tags`)
  - Single file deployment

### To-do:

- Browse for `option.sav` to narrow down selection and prevent user errors.
  - Currently waiting for _me_ to update my Avalonia File Browser extension to support filtering.
  
### Notes:

If you wish to test the built application, there are releases on my [fork](https://github.com/ArchLeaders/BotwSaveManager/releases) with compiled executables.

### CODE REVIEW

**Comment by DeltaJordan** [comment]
Thank you for this. I originally wrote this when I was a lot less experienced than I am now haha, but this is really nicely done. I'll give the code a quick review and approve this if everything checks out.

File: BotwSaveManager.Core/Helpers/Logger.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace BotwSaveManager.Core.Helpers
+{
+    public static class Logger
+    {
+        private static readonly string SourceRoot = "F:\\GitHub\\BotwSaveManager";
```
Comment (DeltaJordan) [inline BotwSaveManager.Core/Helpers/Logger.cs]: Don't use hard-coded paths. If you want to have a logger, use `AppDomain.CurrentDomain.BaseDirectory` to get the current application path, then add a folder for logging.

File: BotwSaveManager.Console/Program.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+﻿
+// 
+// Credits
+using BotwSaveManager.Core;
+using BotwSaveManager.Core.Helpers;
+using System.Diagnostics;
+
+Console.WriteLine("Breath of the Wild Save Manager by Jordan Zeotni, fork of https://github.com/WemI0/BOTW_SaveConv, rewritten and updated by Arch Leaders\n\n");
```
Comment (DeltaJordan) [inline BotwSaveManager.Console/Program.cs]: Update "Jordan Zeotni" to "Jordan Marine" I used to use a fake last name back in 2020. That's my fault.

File: BotwSaveManager/Resource.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public class Resource
+    {
+        internal static readonly string Namespace = typeof(Resource).Namespace ?? "BotwSaveManager";
+        internal static readonly string BaseUrl = $"https://raw.githubusercontent.com/ArchLeaders/{Namespace}/master";
+
+        public byte[] Data { get; set; }
+        public Resource(byte[] data) => Data = data;
+        public Resource(string name, string root)
+        {
+            if (File.Exists($"{root}/{name}")) {
```
Comment (DeltaJordan) [inline BotwSaveManager/Resource.cs]: When combining paths in cross-platform applications, it is better to use `Path.Combine` then to concatenate two paths, as there are a few (uncommon) cases where putting a slash between two paths causes issues.

File: README.md [hunk 4]
```diff
@@ -1,30 +1,40 @@
...
+2. In the second folder selection prompt, browse for an empty folder to save the converted files to.
+3. A debug log will appear and log the converting process.
+4. Once it has completed, and there are no errors, a prompt will appear saying the conversion was successful.
+
+
+### Console Application
+
+1. Drag and drop a WiiU or Switch save folder over the console window as prompted.
+2. Wait for the save to process, then drag an empty folder over the console window to save the converted files to.
+3. Wait once more for the completion message.
```
Comment (DeltaJordan) [inline README.md:40]: Well written description, thanks.

**Review by DeltaJordan — CHANGES_REQUESTED** [review]

File: BotwSaveManager.Core/Helpers/Logger.cs (hunk 1 above)
Comment (ArchLeaders) [reply BotwSaveManager.Core/Helpers/Logger.cs]: This isn't the log outputs folder, it's the path to the source code to shorten the source file paths in the logs. (See [Line 29](https://github.com/DeltaJordan/BotW-Save-Manager/pull/17/files/b42c572cab8c3248be4f15dacae35e4c4dd2450d#diff-0e86ff6cb991092e11c8571f90c4e383ea824b1817e647f04dcd66f113bb357dR29))
However, I may need to change it to reflect the workflow path, as I only tested this when building on my machine (and therefore my source code path).

**Review by DeltaJordan — APPROVED** [review]

**Pull request merged**